### PR TITLE
correct Buildfiles in DataFormats

### PR DESCRIPTION
**Pull request by davidlange6**
Usual automated script for finding missing dependencies..

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

**Comment by makortel** [comment]
+1

**Comment by jfernan2** [comment]
+1

File: DataFormats/RecoCandidate/BuildFile.xml [hunk 1]
```diff
@@ -8,6 +8,11 @@
 <use name="DataFormats/HcalRecHit"/>
 <use name="clhep"/>
 <use name="SimDataFormats/TrackingAnalysis"/>
+<use name="DataFormats/EcalRecHit"/>
```
Comment (perrotta) [inline DataFormats/RecoCandidate/BuildFile.xml:11]: "DataFormats/HcalRecHit", "DataFormats/EcalRecHit", "DataFormats/HcalDetId" are only included inside DataFormats/RecoCandidate/src/classes.h , but actually not needed (and not even included in the classes_def.xml file).
I think one could rather remove them from classes.h, and from this BuildFile

**Comment by smorovic** [comment]
+1

**Comment by civanch** [comment]
+1

File: DataFormats/RecoCandidate/src/classes.h [hunk 2]
```diff
@@ -20,9 +20,6 @@
 #include "DataFormats/RecoCandidate/interface/TrackAssociation.h"
 #include "DataFormats/HcalRecHit/interface/HcalRecHitCollections.h"
```
Comment (perrotta) [inline DataFormats/RecoCandidate/src/classes.h:21]: Also this one can be removed

**Comment by davidlange6** [comment]
please test

**Comment by davidlange6** [comment]
please test

On Mar 9, 2021 8:07 PM, cmsbuild <notifications@github.com> wrote:

+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33096/21471

  *   This PR adds an extra 52KB to repository

**Comment by perrotta** [comment]
+1
- Technical
- Jenkins tests pass

**Comment by srimanob** [comment]
+Upgrade

**Comment by fwyzard** [comment]
Not necessarily for this PR, but can we somehow "sort" the entries in the buidlfiles ?
Like
  - all externals, alphabetically
  - all package dependencies, alphabetically
  - all flags
  - all explicit rules

?

**Comment by jfernan2** [comment]
+1

**Comment by Martin-Grunewald** [comment]
+1

**Comment by smorovic** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by yuanchao** [comment]
+1
* only BuildFile.xml as well as one header file are modified
* matrix tests and unit tests all passed
* static build warnings are not due to the PR

**Comment by silviodonato** [comment]
@cms-sw/core-l2 @cms-sw/l1-l2 can we merge this PR?

**Comment by smuzaffar** [comment]
+core

**Comment by cecilecaillol** [comment]
+l1

**Comment by silviodonato** [comment]
+1

**Pull request merged**